examples/text-generation/run_generation.py

```diff
@@ -177,6 +177,26 @@ def main():
         type=str,
         help="Output directory to store results in.",
     )
+    parser.add_argument(
+        "--bucketsize",
+        default=-1,
+        type=int,
+        help="Bucket size to maintain static shapes. If this number is negative (default is -1) \
+            then we use shape = prompt_length + max_new_tokens. If a positive number is passed \
+            we increate the bucket in steps of bucketsize instead of allocating to max (prompt_length + max_new_tokens)",
+    )
+    parser.add_argument(
+        "--dataset_prompt_length",
+        default=16,
+        type=int,
+        help="Chop dataset to this length. If dataset_prompt_length<=0, do not chop (use full prompt). Default=16"
+    )
```

Comment:
I think it's similar to this PR that just got merged right?
https://github.com/huggingface/optimum-habana/pull/426